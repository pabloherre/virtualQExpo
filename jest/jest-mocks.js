import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock';

const mockfeathersMethods = {
  create: jest.fn(),
  update: jest.fn(),
  find: jest.fn(),
  get: jest.fn(),
  patch: jest.fn(),
  remove: jest.fn()
};
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.requireActual('./mocks');
jest.mock('@react-native-community/async-storage', () => mockAsyncStorage);
jest.mock('expo-barcode-scanner');
jest.mock('expo-location');
jest.mock('react-native-maps');
jest.mock('../src/setup/feathersClient', () => {
  return {
    __esModule: true,
    default: jest.fn(),
    userApi: mockfeathersMethods,
    appointmentApi: mockfeathersMethods,
    businessApi: mockfeathersMethods,
    categoryApi: mockfeathersMethods,
    turnApi: mockfeathersMethods,
    authenticate: jest.fn(),
    reAuthenticate: jest.fn()
  };
});
jest.mock('../src/services/user/User.service');
jest.mock('../src/modules/auth/Auth.service');
jest.mock('../src/common/sideMenu/SideMenu.service');
jest.genMockFromModule('../src/setup/store');
