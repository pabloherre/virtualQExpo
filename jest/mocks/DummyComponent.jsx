import React, { Component } from 'react';
import { View } from 'react-native';

export default class DummyComponent extends Component {
  render() {
    return <View></View>;
  }
}
