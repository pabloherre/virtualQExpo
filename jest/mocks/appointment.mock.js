export const appointment = {
  _id: 1,
  date: new Date("2020-01-01"),
  number: 5,
  status: 'ACTIVE',
  turn: {
    name: 'PAMI',
    status: 'ACTIVE',
    currentTurn: 1,
    code: 'adDweD',
    business: {
      name: 'Business Name',
      category: '5f4d126aeb085b0ef440f066',
      address: 'San Martin 849',
      latitude: -32.3232,
      longitude: -68.6868,
      city: 'Guaymallen',
      state: 'Mendoza'
    }
  }
};
