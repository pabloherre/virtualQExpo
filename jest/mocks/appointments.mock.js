export const appointments = [
  {
    id: 1,
    date: 'Today',
    business: 'Pharmacy',
    appointmentNumber: '234234',
    appointmentCountdown: 23,
    latitude: -32.8913568,
    longitude: -68.8567443
  },
  {
    id: 2,
    date: 'Today',
    business: 'Supermarket',
    appointmentNumber: '2B4354',
    appointmentCountdown: 322,
    latitude: -32.8913568,
    longitude: -68.8567443
  },
  {
    id: 3,
    date: 'Tomorrow',
    business: 'Cinema',
    appointmentNumber: '234354',
    appointmentCountdown: 322,
    latitude: -32.8913568,
    longitude: -68.8567443
  },
  {
    id: 4,
    date: 'Tomorrow',
    business: 'Pool',
    appointmentNumber: '234354',
    appointmentCountdown: 322,
    latitude: -32.8913568,
    longitude: -68.8567443
  }
];
