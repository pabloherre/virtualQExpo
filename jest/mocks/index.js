export * from './appointment.mock';
export * from './appointments.mock';

