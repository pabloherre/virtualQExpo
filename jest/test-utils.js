import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);
function connectedRender(ui, initialState) {
  const store = mockStore(initialState);
  component = renderer.create(<Provider store={store}>{ui}</Provider>);
  return component;
}

// override connectedRender method
export { connectedRender };
