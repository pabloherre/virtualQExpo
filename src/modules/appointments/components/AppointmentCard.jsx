import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Text, TouchableOpacity, View } from 'react-native';
import styles from './AppointmentCard.styles';
import Typography from '../../../common/typography/Typography';
import { AntDesign } from '@expo/vector-icons';
import { withTheme } from '../../../common/theme/Theme';
import { compose } from 'redux';
import { AppointmentDefinition } from '../appointments.definitions';
import { colors } from '../../../../theme';
import { setTestInfo } from '../../../utils/test.utils';

export class AppointmentCard extends Component {
  render() {
    const { appointment, onPressCard } = this.props;
    return (
      <TouchableOpacity style={styles.card} onPress={() => onPressCard && onPressCard(appointment)} {...setTestInfo('testTouchableAppointmentCard')}>
        <View>
          <Typography size="small" color="primary" {...setTestInfo('testDateAppointmentDate')}>
            {appointment.date.toString()}
          </Typography>
          <Typography {...setTestInfo('testTextAppointmentBusiness')} textStyles={{ fontWeight: 'bold' }}>
            {appointment.turn.business.name}
          </Typography>
        </View>

        <View>
          <Text>&nbsp;</Text>
          <Typography size="small">Your #</Typography>
          <View style={{ flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between' }}>
            <Typography size="large" color="secondary" {...setTestInfo('testTextAppointmentNumber')}>
              {appointment.number}
            </Typography>
            <AntDesign name="right" size={24} color={colors.text} />
            <Typography size="large" color="primary" {...setTestInfo('testTextAppointmentNumberAhead')}>
              {appointment.number - appointment.turn.currentTurn}
            </Typography>
            <Typography size="small" textStyles={{ width: 60 }}>
              Numbers ahead
            </Typography>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}

AppointmentCard.propTypes = {
  appointment: PropTypes.shape(AppointmentDefinition).isRequired
};

function mapStateToProps(state) {
  return {};
}

export default compose(withTheme, connect(mapStateToProps))(AppointmentCard);
