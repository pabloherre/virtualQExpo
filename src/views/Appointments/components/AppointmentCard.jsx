import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { TouchableOpacity, View, Text } from 'react-native';
import styles from './AppointmentCard.styles';
import Typography from '../../../common/typography/Typography';

class AppointmentCard extends Component {
  render() {
    const { appointment } = this.props;
    return (
      <TouchableOpacity style={styles.card}>
        <View>
          <Typography size="small" color="primary">
            {appointment.date}
          </Typography>
          <Typography textStyles={{ fontWeight: 'bold' }}>{appointment.business}</Typography>
        </View>

        <View>
          <Text>&nbsp;</Text>
          <Typography size="small">Your #</Typography>
          <View style={{ flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between' }}>
            <Typography size="large" color="secondary">
              {appointment.appointmentNumber}
            </Typography>
            <Typography size="large">&#62;</Typography>
            <Typography size="large" color="primary">
              {appointment.appointmentCountdown}
            </Typography>
            <Typography size="small" textStyles={{ width: 60 }}>
              Numbers ahead
            </Typography>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}

AppointmentCard.propTypes = {
  appointment: PropTypes.shape({
    appointmentNumber: PropTypes.string.isRequired,
    business: PropTypes.string.isRequired,
    appointmentCountdown: PropTypes.number,
    date: PropTypes.date
  })
};

function mapStateToProps(state) {
  return {};
}

export default connect(mapStateToProps)(AppointmentCard);
