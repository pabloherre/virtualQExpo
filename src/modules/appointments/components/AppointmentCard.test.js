import React from 'react';
import { AppointmentCard } from './AppointmentCard';
import { shallow } from 'enzyme';
import { appointment } from '../../../../jest/mocks';

describe('<AppointmentCard />', () => {
  let wrapper;
  const onPressCard = jest.fn();

  beforeAll(() => {
    wrapper = shallow(<AppointmentCard appointment={appointment} onPressCard={onPressCard} />);
  });

  afterEach(() => {
    onPressCard.mockClear();
  });

  it('should render correctly', async () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have date', async () => {
    const el = wrapper.find("[testID='testDateAppointmentDate']");
    expect(el.childAt(0).text()).toBe(new Date('2020-01-01').toString());
  });

  it('should have business name', async () => {
    const el = wrapper.find("[testID='testTextAppointmentBusiness']");
    expect(el.childAt(0).text()).toBe('Business Name');
  });

  it('should have number', async () => {
    const el = wrapper.find("[testID='testTextAppointmentNumber']");
    expect(el.childAt(0).text()).toBe('5');
  });

  it('should have {n} numbers ahead', async () => {
    const el = wrapper.find("[testID='testTextAppointmentNumberAhead']");
    expect(el.childAt(0).text()).toBe('4');
  });

  it('should call function on touch', async () => {
    const el = wrapper.find("[testID='testTouchableAppointmentCard']");
    el.simulate('press');
    expect(onPressCard).toHaveBeenCalled();
  });
});
