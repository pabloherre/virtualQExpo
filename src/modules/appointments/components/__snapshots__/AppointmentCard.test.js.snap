// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentCard /> should render correctly 1`] = `
<ForwardRef
  accessibilityLabel="testTouchableAppointmentCard"
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "#e7eefb",
      "borderRadius": 10,
      "flex": 1,
      "margin": 5,
      "padding": 20,
    }
  }
  testID="testTouchableAppointmentCard"
>
  <View>
    <Component
      accessibilityLabel="testDateAppointmentDate"
      color="primary"
      size="small"
      testID="testDateAppointmentDate"
    >
      Tue Dec 31 2019 21:00:00 GMT-0300 (Argentina Standard Time)
    </Component>
    <Component
      accessibilityLabel="testTextAppointmentBusiness"
      testID="testTextAppointmentBusiness"
      textStyles={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Business Name
    </Component>
  </View>
  <View>
    <Text>
      Â 
    </Text>
    <Component
      size="small"
    >
      Your #
    </Component>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Component
        accessibilityLabel="testTextAppointmentNumber"
        color="secondary"
        size="large"
        testID="testTextAppointmentNumber"
      >
        5
      </Component>
      <Icon
        allowFontScaling={false}
        color="#a1a2a4"
        name="right"
        size={24}
      />
      <Component
        accessibilityLabel="testTextAppointmentNumberAhead"
        color="primary"
        size="large"
        testID="testTextAppointmentNumberAhead"
      >
        4
      </Component>
      <Component
        size="small"
        textStyles={
          Object {
            "width": 60,
          }
        }
      >
        Numbers ahead
      </Component>
    </View>
  </View>
</ForwardRef>
`;
